src/main/kotlin/ecommerce/api/ProductController.kt

```diff
@@ -0,0 +1,57 @@
+package ecommerce.api
+
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.DeleteMapping
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.PutMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+import java.net.URI
+
+@RestController
+@RequestMapping("/api")
+class ProductController(private val productService: ProductService) {
+    @PostMapping("/products")
+    fun createProduct(
+        @RequestBody product: Product,
+    ): ResponseEntity<Void> {
```

Comment:
```suggestion
    ): ResponseEntity<Unit> {
```
This is not a must, but we can actually use `Unit` here as well.
What are the differences?
+) after looking up the differences, what would you use?

Reply:
Interesting. Kotlin `Unit` corresponds to the `void` in Java. Since Spring Boot is a Java project, and I suppose that my project is going to be processed through JVM, so I might be down to `ResponseEntity<Void>`.